backend/database.js

```diff
@@ -380,42 +402,53 @@ const appendToNamesIndex = appendToIndexFactory('_private_names_index')
  * @param key - The key that identifies the index in the database.
  * @returns A function that takes a value to remove from the index.
  */
-export const removeFromIndexFactory = (key: string): (value: string) => Promise<void> => {
-  return (value: string) => {
+export const removeFromIndexFactory = (key: string): (values: string | string[]) => Promise<void> => {
```

Comment:
Does it make sense to have this sort of asymmetry between the APIs? (`append` takes only a single value, whereas `remove` can take an array? Or is there just never a need for an array with `append`?

Reply:
There hasn't been a need for an array with append, as we always add things one-by-one.